zrml/swaps/rpc/src/lib.rs

```diff
@@ -75,6 +76,14 @@ where
         with_fees: bool,
         blocks: Vec<BlockNumber>,
     ) -> RpcResult<Vec<SerdeWrapper<Balance>>>;
+
+    #[method(name = "swaps_getAssetSpotPricesForPool")]
+    async fn get_asset_spot_prices_for_pool(
```

Comment:
I'd say this needs a different name. Otherwise the difference to the function `get_spot_prices` is unclear. After all, that function also gets spot prices of assets which are in a pool. I suggest `get_spot_prices_all` or `get_all_spot_prices`. Open to suggestions.